sdk/how-to-guides/getting-started-with-testing.md

```diff
@@ -17,97 +16,85 @@ What you will learn:
 
-## Charmcraft
-The template you get with `charmcraft init` is rather compact. The testing infrastructure consists of a `tests/test_charm.py` file, and a `run_tests` shell script that contains:
-```bash
-                                                      
-if [ -z "$VIRTUAL_ENV" -a -d venv/ ]; then            
-    . venv/bin/activate                               
-fi                                                    
-                                                      
-if [ -z "$PYTHONPATH" ]; then                         
-    export PYTHONPATH="lib:src"                       
-else                                                  
-    export PYTHONPATH="lib:src:$PYTHONPATH"           
-fi                                                    
-                                                      
-flake8                                                
-coverage run --branch --source=src -m unittest -v "$@"
-coverage report -m                                    
+```shell
+charmcraft init --profile=machine
 ```
-In words, this will run:
- - `flake8` linter checks.
- - `unittest` with coverage in `/tests/`.
 
-## Template-operator
-The `template-operator` template provides a more robust starting point for most charming needs. 
-It comes with:
- - example unit tests (`/tests/unit/test_charm.py`) 
- - example integration tests (`/tests/integration/test_charm.py`)
- - instead of a `run_tests` script, a `tox.ini` file that contains pre-set environments for:
-    - formatting (`black`)
-    - linting (`flake8`, `isort`)
-    - unit (`pytest`)
-    - integration (`pytest`, `pytest-operator`)
+This will provide the following files that you'll use when writing your tests (as well as others for building your charm):
 
-So `template-operator` contains all of `charmcraft init`'s template tooling, and more. It also contains github actions specifications that will configure GH to run the whole suite on each PR to `master/main` (more on that later).
+```
+.
+├── pyproject.toml
+├── spread.yaml
+├── tests
+│   ├── integration
+│   │   └── test_charm.py
+│   ├── spread
+│   │   ├── general
+│   │   │   └── integration
+│   │   │       └── task.yaml
+│   │   └── lib
+│   │       ├── cloud-config.yaml
+│   │       └── test-helpers.sh
+│   └── unit
+│       └── test_charm.py
+└── tox.ini
+```
 
-[note]`template-operator` is in practice a k8s charm template. Its example code is that of a kubernetes charm, and its CI will run the integration tests on `microk8s`. If you're developing a machine charm, you are going to have to change all that.[/note]
+There are also profiles for `kubernetes` and for building charms for apps developed with popular frameworks such as Django and Flask.
 
-In the rest of this tutorial we will assume that our project is set up with `template-operator`.
+> See more: [ Write your first Kubernetes charm for a Flask app](https://juju.is/docs/sdk/write-your-first-kubernetes-charm-for-a-flask-app)
 
```

Comment:
We have all of this info on charmcraft init and profiles here: https://juju.is/docs/sdk/set-up-a-charm-project 

If we're missing details, we should add them there rather than trying to redefine here.

Reply:
Same as above, it currently has a bunch of content from the outdated template repo, so this was an attempt at a 1-to-1 swap out.

(I do agree that when this is refactored/rethought it shouldn't have this duplication).